base/task.jl

```diff
@@ -121,6 +121,29 @@ Get the currently running [`Task`](@ref).
 """
 current_task() = ccall(:jl_get_current_task, Ref{Task}, ())
 
+# task states
+
+const task_state_runnable = UInt8(0)
+const task_state_done     = UInt8(1)
+const task_state_failed   = UInt8(2)
+
+@inline function getproperty(t::Task, field::Symbol)
+    if field === :state
```

Comment:
Is the intent to add a depwarn here in v2? If so, perhaps a comment would be useful?